### PR TITLE
[IOS-1944] Conviva integration demo app

### PR DESCRIPTION
**Pull request by kobertynska**
- Conviva integration best practice app is added.
- JWPlayer is updated to 3.2.0 version.
- Other best practice apps are updated according to new version of JWPlayer.

### CODE REVIEW

File: JWBestPracticeApps/Podfile [hunk 1]
```diff
@@ -1,31 +1,35 @@
 # Uncomment this line to define a global platform for your project
-# platform :ios, '8.0'
+# platform :ios, '11.0'
 # Uncomment this line if you're using Swift
 # use_frameworks!
 
+def common_JW
+    pod 'JWPlayer-SDK', '~> 3.2.0'
```
Comment (karimMourra) [inline JWBestPracticeApps/Podfile]: wouldn't this limit us to only 3.2.x ? We will have to change this for 3.3.0 right ?
let's use `pod 'JWPlayer-SDK', '~> 3.0'`

File: JWBestPracticeApps/JWConviva/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    
+    var analyticsObserver: AnalyticsObserver?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        analyticsObserver = AnalyticsObserver(player: player)
+        player.analyticsDelegate = analyticsObserver
+        
+        if let playerView = player.view {
```
Comment (karimMourra) [inline JWBestPracticeApps/JWConviva/ViewController.swift]: this code is not needed for Conviva, correct ? If it is best practice for general setup, it should go in the basic demo https://github.com/jwplayer/jwplayer-ios-bestPracticeApps/blob/da0a8f881b123a70f37edde540c64302cdeb14e3/JWBestPracticeApps/JWBestPracticeApps/JWBasicVideoViewController.m

File: JWBestPracticeApps/Podfile (hunk 1 above)
Comment (kobertynska) [reply JWBestPracticeApps/Podfile]: @karimMourra It's changed.

File: JWBestPracticeApps/JWConviva/ViewController.swift (hunk 2 above)
Comment (kobertynska) [reply JWBestPracticeApps/JWConviva/ViewController.swift]: @karimMourra player setup is removed from Conviva example.

**Comment by karimMourra** [comment]
@kobertynska could you please add a readme to the conviva target ? All the other targets have a readme, you can use those as examples

File: JWBestPracticeApps/JWConviva/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//  ViewController.swift
+//  JWConviva
+//
+//  Created by Kateryna Obertynska on 3/18/19.
+//  Copyright © 2019 Karim Mourra. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: JWBasicVideoViewController {
```
[the remark this answers is not in the document]
Comment (kobertynska) [reply JWBestPracticeApps/JWConviva/ViewController.swift]: @karimMourra done

File: JWBestPracticeApps/JWConviva/AnalyticsObserver.swift [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+//  Copyright © 2019 JWPlayer. All rights reserved.
+//
+
+import Foundation
+import AVFoundation
+import ConvivaSDK
+
+let TEST_CUSTOMER_KEY = "CUSTOMER_KEY"
+let TOUCHSTONE_SERVICE_URL = "https://optus-test.testonly.conviva.com/"
+
```
[the remark this answers is not in the document]
Comment (kobertynska) [reply JWBestPracticeApps/JWConviva/AnalyticsObserver.swift]: @karimMourra done

**Comment by kobertynska** [comment]
@karimMourra README file is added. Feel free to edit it.

File: JWBestPracticeApps/JWConviva/AnalyticsObserver.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (karimMourra) [reply JWBestPracticeApps/JWConviva/AnalyticsObserver.swift]: @okulagin good catch!

File: JWBestPracticeApps/JWConviva/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+The JWConviva target illustrates how to use 3rd party analytics libraries such as Conviva. The example provides the understanding of how to create an analytics observer and pass video and ads data to the Conviva library. Please note that you need to set your key and service url in the code in order to see the tracked data within the Conviva dashboard.
```
Comment (karimMourra) [inline JWBestPracticeApps/JWConviva/README.md]: @kobertynska  Excellent description!

File: JWBestPracticeApps/JWConviva/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+
+Please refer to the Conviva documentation to get understanding of what data can be collected by the library so that you can develop your own custom analytics listener that matches your needs.
```
Comment (AmitaiB) [inline JWBestPracticeApps/JWConviva/README.md]: "...to get **_an_** understanding of..."

**Comment by AmitaiB** [comment]
@kobertynska  In updating the Best Practice Apps, this also satisfies [IOS-1909](https://jwplayer.atlassian.net/browse/IOS-1909) and (if/when taken together with updating the demo app) [SERV-1000](https://jwplayer.atlassian.net/browse/SERV-1000).

**Pull request merged**